src/main/java/org/openrewrite/java/template/RefasterTemplateProcessor.java

```diff
@@ -257,6 +257,17 @@ public void visitClassDef(JCTree.JCClassDecl classDecl) {
                             recipe.append("                    return elem;\n");
                             recipe.append("                }\n");
                         }
+
+                        for (String import_ : imports) {
+                            recipe.append("                maybeRemoveImport(\"" + import_ + "\");\n");
+                            recipe.append("                maybeAddImport(\"" + import_ + "\");\n");
+                        }
+                        for (String import_ : staticImports) {
+                            recipe.append("                maybeRemoveImport(\"" + import_ + "\");\n");
+                            int dot = import_.lastIndexOf('.');
+                            recipe.append("                maybeAddImport(\"" + import_.substring(0, dot) + "\", \"" + import_.substring(dot + 1) + "\");\n");
+                        }
```

Comment:
I was surprised we don't seem to separate imports, even though they end up OK into each of the before/after templates. That meant here I've had to both add and remove them _maybe_; which was good enough to get working, but not something that I'd be confident relying on.